infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -499,25 +530,35 @@ private void analyzePollReportAndTriggerHandler(PollReport pollReport) {
                 // Failures detected - unresponsive servers.
                 // We check if these servers are the same set of servers which are marked as
                 // unresponsive in the layout. If yes take no action. Else trigger handler.
-                log.info("Failures detected. Failed nodes : {}", pollReport.toString());
                 // Check if this failure has already been recognized.
                 //TODO: Does not handle the un-marking case where markedSet is a superset of
                 // pollFailures.
                 for (String failedServer : pollReport.getFailingNodes()) {
                     if (!latestLayout.getUnresponsiveServers().contains(failedServer)) {
+                        log.info("Failures detected. Failed nodes : {}", pollReport.toString());
```

Comment:
These if conditions should move to small testable methods and they should build a set of failing nodes with one action at the end.

Reply:
Done